packages/protocol/contracts/dapis/RrpDapiServer.sol

```diff
@@ -28,13 +38,45 @@ contract RrpDapiServer is MeanMedianHybrid {
 
   AirnodeRrp public airnodeRrp;
   mapping(bytes16 => Dapi) private dapis;
+  // TODO: are we missing a dapiIds bytes16 array for getting all the dapi ids?
   uint256 public nextDapiIndex = 1;
   mapping(bytes32 => DapiRequestIdentifiers) private airnodeRequestIdToDapiRequestIdentifiers;
 
   constructor(address _airnodeRrp) {
     airnodeRrp = AirnodeRrp(_airnodeRrp);
   }
 
+  function getTemplateIdIndex(bytes16 _dapiId, bytes32 _templateId) external view returns (uint256, bool) {
+    return (dapis[_dapiId].templateIdToIndex[_templateId].index, dapis[_dapiId].templateIdToIndex[_templateId].exists);
+  }
+
+  function getDapi(bytes16 dapiId)
+    external
+    view
+    returns (
+      uint256,
+      uint256,
+      uint256,
+      bytes32[] memory,
+      address[] memory,
+      address,
+      bytes4,
+      address
+    )
+  {
+    Dapi storage dapi = dapis[dapiId];
```

Comment:
pardon my ignorance, by why does this need to be storage?

Reply:
My understanding is that storage means "pointer to" so in this case I am creating a pointer to the dapi struct in state instead of making a copy because I just want to read it's values and return them.